latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,233 @@
+        "workdir /root",
+    ]
+
+
+def infer_commands(pkg_root: Path) -> List[DockerCmdBlock]:
+    commands: List[DockerCmdBlock] = []
+
+    if (pkg_root / "system-requirements.txt").exists():
+        print("Adding system requirements installation phase")
+        commands.append(
+            DockerCmdBlock(
+                comment="install system requirements",
+                commands=[
+                    "copy system-requirements.txt /opt/latch/system-requirements.txt",
+                    "run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            )
+        )
+
+    if (pkg_root / "environment.R").exists():
+        print("Adding R + R package installation phase")
+        commands.append(
+            DockerCmdBlock(
+                comment="install R requirements",
+                commands=[
+                    dedent(
+                        r"""
+                        run apt-get update --yes && \
+                            apt-get install --yes software-properties-common && \
+                            add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \
+                            apt-get install --yes r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget
+                        """
+                    ).strip(),
+                    "copy environment.R /opt/latch/environment.R",
+                    "run Rscript /opt/latch/environment.R",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            )
+        )
+
+    conda_env_file_name = None
+    if (pkg_root / "environment.yml").exists():
+        conda_env_file_name = "environment.yml"
+    elif (pkg_root / "environment.yaml").exists():
+        conda_env_file_name = "environment.yaml"
+
+    if conda_env_file_name is not None:
+        print("Adding conda + conda environment installation phase")
+        with (pkg_root / conda_env_file_name).open("rb") as f:
+            conda_env = yaml.safe_load(f)
+
+        if "name" in conda_env:
+            env_name = conda_env["name"]
+        else:
+            env_name = "workflow"
+
+        commands += [
+            DockerCmdBlock(
+                comment="set conda environment variables",
+                commands=[
+                    "env CONDA_DIR /opt/conda",
+                    "env PATH=$CONDA_DIR/bin:$PATH",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+            DockerCmdBlock(
+                comment="install miniconda",
+                commands=[
+                    dedent(
+                        r"""
+                        run apt-get update --yes && \
+                            apt-get install --yes curl && \
+                            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
```

Comment:
```suggestion
                            curl --remote-name https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
```

also a nit but just fyi `--output` is `-o` not `-O` (see https://curl.se/docs/manpage.html#-O and https://curl.se/docs/manpage.html#-o)